html/admin/api.php

```diff
@@ -0,0 +1,56 @@
+<?php
+header('Content-Type: application/json; charset=UTF8');
+define("_CMD", "sudo /usr/local/biblebox/bin/BBoxManage.sh");
+$method = $_SERVER['REQUEST_METHOD'];
+$path_info = explode('/', trim($_SERVER['PATH_INFO'],'/'));
+$request_body = json_decode(file_get_contents('php://input'),true);
+
+if ( count( $path_info ) < 1 ) {
+    http_response_code(404);
+    die('NOT FOUND');
+}
+
+$property = $path_info[0];
+
+$valid_properties = array("ssid", "channel", "hostname", "password", "system");
+if ( in_array($property, $valid_properties) ) {
+    if ($method == 'GET') {
```

Comment:
So, the input I am calling property in the script is whitelisted; you can only input a particular one specified in valid_properties. For the put request, where we have user input, the escapeShellCommand function is used to sanitize it. I've not done a lot of php, though, so it would probably be good to have someone who has review it for security wholes.